src/main/java/com/artipie/asto/blocking/BlockingStorage.java

```diff
@@ -43,30 +42,25 @@
     /**
      * Wrapped storage.
      */
-    private final Storage storage;
+    private final RxStorage storage;
```

Comment:
@Sammers21 could you please explain why we need to use `RxStorage` here? It will be way more simple to just keep using normal `Storage`, but to invoke `.join()` on it's results instead of `.get()`

Reply:
@olegmoz, because `join` is a bad name for the operation. `CompletableFuture` in general is a badly-designed primitive for reactive programming and should be avoided where possible.

Reply:
@Sammers21 sorry, can't agree with you on `join` name being bad. It resembles "join thread" I believe. And `CompletableFuture` is not reactive programming primitive. Though it does not matter, neither `Storage` nor `BlockingStorage` are reactive and there is no sense to put something "reactive" in-between: it's unnecessary complication of code and waste of computation resources.